Collision handler y collision layers

Auraum opened the pull request:
[Finishes #171234302]
Colisión entre jugadores y otros objetos (para dañar al jugador) implementada de manera provisional como ejemplo de manejo de colisiones. Pick ups y grab se añadirán en el handler cuando estén.
También añadidos collision layers de verdad que funcionan por enum.

Auraum committed:
Empieza el collision handler

Auraum committed:
Collision handler y collision layers

[Finishes #171234302]
Colisión entre jugadores y otros objetos (para dañar al jugador) implementada de manera provisional como ejemplo de manejo de colisiones. Pick ups y grab se añadirán en el handler cuando estén.
También añadidos collision layers de verdad que funcionan por enum.

JuanmaSalva reviewed (approved):
Parece bastante robusto aun

EstebenR committed:
Cambios de limpieza

Cambiado 'Triggers' por 'Trigger' porque me triggereaba.

Cambiado orden de varias clases

Añadido el collisionHandler_ como atributo a PlayState

EstebenR committed:
Update Collider.cpp

AndresRuizB reviewed (approved):
venga va
